ipykernel/jsonutil.py

```diff
@@ -83,6 +86,13 @@ def encode_images(format_dict):
         if not jpegdata.startswith(JPEG64):
             jpegdata = encodebytes(jpegdata)
         encoded['image/jpeg'] = jpegdata.decode('ascii')
+        
+    gifdata = format_dict.get('image/gif')
+    if isinstance(gifdata, bytes):
+        # make sure we don't double-encode
+        if not gifdata.startswith(GIF) or not gifdata.startswith(GIF89):
```

Comment:
When you generate the base64 encoded versions, let's grab a few different GIFs and base64 encode them just to check that the prefixes are accurate, as well.